v0/destinations/snapchat_custom_audience/transform.js

```diff
@@ -0,0 +1,188 @@
+  CustomError
+} = require("../../util");
+const ErrorBuilder = require("../../util/error");
+const { BASE_URL } = require("./config");
+
+/**
+ * Get access token to be bound to the event req headers
+ *
+ * Note:
+ * This method needs to be implemented particular to the destination
+ * As the schema that we'd get in `metadata.secret` can be different
+ * for different destinations
+ *
+ * @param {Object} metadata
+ * @returns
+ */
+const getAccessToken = metadata => {
+  // OAuth for this destination
+  const { secret } = metadata;
+  // we would need to verify if secret is present and also if the access token field is present in secret
+  if (!secret || !secret.access_token) {
+    throw new ErrorBuilder()
+      .setStatus(500)
+      .setMessage("Empty/Invalid access token")
+      .build();
+  }
+  return secret.access_token;
+};
+
+/**
+ * Verifies whether the input payload is in right format or not
+ * @param {Object} message
+ * @returns
+ */
+const validatePayload = message => {
+  if (!message.type) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "[snapchat_custom_audience]::Message Type is not present. Aborting message."
+      )
+      .setStatus(400)
+      .build();
+  }
+  if (!message.properties) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "[snapchat_custom_audience]::Message properties is not present. Aborting message."
+      )
+      .setStatus(400)
+      .build();
+  }
+  if (!message.properties.listData) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "[snapchat_custom_audience]::listData is not present inside properties. Aborting message."
+      )
+      .setStatus(400)
+      .build();
+  }
+  if (message.type.toLowerCase() !== "audiencelist") {
+    throw new ErrorBuilder()
+      .setMessage(
+        `[snapchat_custom_audience]::Message Type ${message.type} not supported.`
+      )
+      .setStatus(400)
+      .build();
+  }
+  if (!message.properties.listData.add && !message.properties.listData.remove) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "[snapchat_custom_audience]::Neither 'add' nor 'remove' property is present inside 'listData'. Aborting message."
+      )
+      .setStatus(400)
+      .build();
+  }
+};
+
+const responseBuilder = (metadata, message, { Config }, type) => {
+  const { schema, segmentId } = Config;
+  const payload = { users: [] };
+  const userPayload = { schema: [], data: [] };
+  let schemaType;
+  let hashedProperty;
+  // Normalizing and hashing ref: https://marketingapi.snapchat.com/docs/#normalizing-hashing
+
+  let userArray;
+  if (type === "add") {
+    userArray = message.properties.listData.add;
+  } else if (type === "remove") {
+    userArray = message.properties.listData.remove;
+  }
+
+  switch (schema) {
+    case "email":
+      schemaType = "EMAIL_SHA256";
+      userArray.forEach(element => {
+        hashedProperty = element?.email;
+        // what if email is not present in all cases?
```

Comment:
what do you mean by this statement. Is it possible to add more than one property in a single element? Like email and phone in one?